### PR TITLE
Terraform Plan and Apply GitHub Action Integrations 

### PR DESCRIPTION
**Pull request by milldr**
## what
- Added documentation pages for GitHub Actions for Terraform Plan and Apply

## why
- Documented the two new GitHub actions for Terraform Plan and Apply

## references
- n/a

### CODE REVIEW

File: website/docs/integrations/github-actions/atmos-terraform-apply.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---
+title: Atmos Terraform Apply
+sidebar_position: 50
+sidebar_label: Terraform Apply
+---
+
+The Cloud Posse GitHub Action "Atmos Terraform Apply" runs Terraform entirely from GitHub Action workflows.
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-apply.md:7]:
```suggestion
The Cloud Posse GitHub Action for "Atmos Terraform Apply" runs Terraform entirely from GitHub Action workflows.
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
+---
+title: Atmos Terraform Plan
+sidebar_position: 40
+sidebar_label: Terraform Plan
+---
+
+The Cloud Posse GitHub Action "Atmos Terraform Plan" runs Terraform entirely from GitHub Action workflows.
```
Comment (milldr) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:7]:
```suggestion
The Cloud Posse GitHub Action for "Atmos Terraform Plan" runs Terraform entirely from GitHub Action workflows.
```

File: website/docs/integrations/github-actions/atmos-terraform-apply.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+Given any component and stack in an Atmos supported infrastructure environment, [`github-action-atmos-terraform-apply`](https://github.com/cloudposse/github-action-atmos-terraform-apply) will retrieve an existing Terraform planfile from a given S3 bucket using metadata in a DynamoDB table, run `atmos terraform apply` with that planfile, and format the Terraform Apply result as part of a GitHub Workflow Summary.
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-apply.md:9]:
```suggestion
Given any component and stack in an Atmos supported infrastructure environment, [`github-action-atmos-terraform-apply`](https://github.com/cloudposse/github-action-atmos-terraform-apply) will retrieve an existing Terraform [planfile](https://developer.hashicorp.com/terraform/tutorials/automation/automate-terraform) from a given S3 bucket using metadata stored inside a DynamoDB table, run `atmos terraform apply` with that planfile, and format the Terraform Apply result as part of a [GitHub Workflow Job Summary](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/).
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+```
+
+### Requirements
+
+This GitHub Action expects an S3 bucket, DynamoDB table, and two access roles. 
+
+#### S3 Bucket
+
+Any basic S3 Bucket can be connected to this action. For example, using Atmos stack configuration, define a bucket using the [`s3-bucket` component](https://github.com/cloudposse/terraform-aws-components/tree/main/modules/s3-bucket) with this catalog configuration:
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:57]:
```suggestion
Any basic S3 Bucket can be connected to this action. For example, [vendor in](/core-concepts/components/vendoring) the [`s3-component`](https://docs.cloudposse.com/components/library/aws/s3-bucket/), then using [Atmos stack configuration](/core-concepts/stacks/), define a bucket using the [`s3-bucket` component](https://github.com/cloudposse/terraform-aws-components/tree/main/modules/s3-bucket) with this catalog configuration:
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
+---
+title: Atmos Terraform Plan
+sidebar_position: 40
+sidebar_label: Terraform Plan
+---
+
+The Cloud Posse GitHub Action for "Atmos Terraform Plan" runs Terraform entirely from GitHub Action workflows.
+
+Given any component and stack in an Atmos supported infrastructure environment, [`github-action-atmos-terraform-plan`](https://github.com/cloudposse/github-action-atmos-terraform-plan) will run `atmos terraform plan`, generate a Terraform planfile, store this planfile in a S3 Bucket with metadata in DynamodDB, and finally format the Terraform Plan result as part of a GitHub Workflow Summary.
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:9]:
```suggestion
Given any component and stack in an Atmos supported infrastructure environment, [`github-action-atmos-terraform-plan`](https://github.com/cloudposse/github-action-atmos-terraform-plan) will run `atmos terraform plan`, generate a Terraform [planfile](https://developer.hashicorp.com/terraform/tutorials/automation/automate-terraform), store this planfile in an S3 Bucket with metadata in DynamodDB, and finally format the Terraform Plan result as part of a [GitHub Workflow Job Summary](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/).
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+          - dynamodb/defaults
+      vars:
+        name: gitops-plan-storage
+        # These keys (case-sensitive) are required for the cloudposse/github-action-terraform-plan-storage action
+        hash_key: id
+        range_key: createdAt
+```
+
+
+Assign this table ARN to the `terraform-plan-table` input.
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:103]:
```suggestion
Assign this table ARN to the `terraform-plan-table` input of the `cloudposse/github-action-atmos-terraform-plan` GitHub Action.
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+#### IAM Access Roles
+
+First create an access role for storing and retrieving planfiles from the S3 Bucket and DynamoDB table. Assign this role ARN to the `terraform-state-role` input.
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:107]: How to do this?

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        name: gitops-plan-storage
+        allow_encrypted_uploads_only: false
+```
+
+
+Assign this S3 Bucket ARN to the `terraform-plan-bucket` input.
+
+#### DynamoDB Table
+
+Similiarly, a basic DynamoDB table can be created with our [`dynamodb` component](https://github.com/cloudposse/terraform-aws-components/tree/main/modules/dynamodb). Set the Hash Key and Range Key as follows:
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:81]:
```suggestion
Similarly, a simple DynamoDB table can be provisioned using our [`dynamodb` component](https://github.com/cloudposse/terraform-aws-components/tree/main/modules/dynamodb). Set the **Hash Key** and **Range Key** as follows:
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+### Requirements
+
+This GitHub Action expects an S3 bucket, DynamoDB table, and two access roles. 
+
+#### S3 Bucket
+
+This action can use any S3 Bucket to keep track of your planfiles. Just ensure the bucket is properly locked down since planfiles may contain secrets.
+
+For example, [vendor in](/core-concepts/components/vendoring) the [`s3-component`](https://docs.cloudposse.com/components/library/aws/s3-bucket/), then using [Atmos stack configuration](/core-concepts/stacks/), define a bucket using the [`s3-bucket` component](https://github.com/cloudposse/terraform-aws-components/tree/main/modules/s3-bucket) with this catalog configuration:
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:59]:
```suggestion
For example, [vendor in](/core-concepts/components/vendoring) the [`s3-component`](https://docs.cloudposse.com/components/library/aws/s3-bucket/), then using an [Atmos stack configuration](/core-concepts/stacks/), define a bucket using the [`s3-bucket` component](https://github.com/cloudposse/terraform-aws-components/tree/main/modules/s3-bucket) with this catalog configuration:
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+        name: gitops-plan-storage
+        allow_encrypted_uploads_only: false
+```
+
+
+Assign this S3 Bucket ARN to the `terraform-plan-bucket` input.
+
+#### DynamoDB Table
+
+Similarly, a simple DynamoDB table can be provisioned using our [`dynamodb` component](https://github.com/cloudposse/terraform-aws-components/tree/main/modules/dynamodb). Set the **Hash Key** and **Range Key** as follows:
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:83]:
```suggestion
Similarly, a simple DynamoDB table can be provisioned using our [`dynamodb` component](https://docs.cloudposse.com/components/library/aws/dynamodb/). Set the **Hash Key** and **Range Key** as follows:
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md (hunk 7 above)
Comment (milldr) [reply website/docs/integrations/github-actions/atmos-terraform-plan.md]: added note, but link depends on https://github.com/cloudposse/terraform-aws-components/pull/735

File: website/docs/integrations/github-actions/atmos-terraform-apply.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+sidebar_position: 50
+sidebar_label: Terraform Apply
+---
+
+The Cloud Posse GitHub Action for "Atmos Terraform Apply" runs Terraform entirely from GitHub Action workflows.
+
+Given any component and stack in an Atmos supported infrastructure environment, [`github-action-atmos-terraform-apply`](https://github.com/cloudposse/github-action-atmos-terraform-apply) will retrieve an existing Terraform [planfile](https://developer.hashicorp.com/terraform/tutorials/automation/automate-terraform) from a given S3 bucket using metadata stored inside a DynamoDB table, run `atmos terraform apply` with that planfile, and format the Terraform Apply result as part of a [GitHub Workflow Job Summary](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/).
+
+This action is intended to be used with [Atmos Terraform Plan](/integrations/github-actions/atmos-terraform-plan)
+
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-apply.md:12]:
```suggestion
This action is intended to be used together with [Atmos Terraform Plan](/integrations/github-actions/atmos-terraform-plan)

## Features

This GitHub Action incorporates superior GitOps support for Terraform by utilizing the capabilities of Atmos, enabling efficient management of large enterprise-scale environments.

* **Implements Native GitOps** with Atmos and Terraform
* **No hardcoded credentials.** Use GitHub OIDC to assume roles.
* **Compatible with GitHub Cloud & Self-hosted Runners** for maximum flexibility. 
* **Beautiful Job Summaries** don't clutter up pull requests with noisy GitHub comments
* **100% Open Source with Permissive APACHE2 License** means you have no expensive subscriptions or long-term commitments.




```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+          terraform-state-bucket: "acme-core-ue2-auto-gitops"
+          terraform-state-role: "arn:aws:iam::999999999999:role/acme-core-ue2-auto-gitops-gha"
+          terraform-state-table: "acme-core-ue2-auto-gitops"
+          aws-region: "us-east-2"
+
+```
+
+### Requirements
+
+This GitHub Action expects an S3 bucket, DynamoDB table, and two access roles.
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:67]:
```suggestion
This GitHub Action depends on a few resources:
* **S3 bucket** for storing planfiles
* **DynamoDB table** for retrieving metadata about planfiles
* **2x IAM roles** for "planning" and accessing the "state" bucket
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+          - dynamodb/defaults
+      vars:
+        name: gitops-plan-storage
+        # These keys (case-sensitive) are required for the cloudposse/github-action-terraform-plan-storage action
+        hash_key: id
+        range_key: createdAt
+```
+
+
+Pass the ARN of this table as the input to the `terraform-plan-table` of the `cloudposse/github-action-atmos-terraform-plan` GitHub Action.
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:119]:
```suggestion
Pass the ARN of this table as the input to the `terraform-plan-table` of the [`cloudposse/github-action-atmos-terraform-plan`](https://github.com/cloudposse/github-action-atmos-terraform-plan) GitHub Action.
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+          stack: "plat-ue2-sandbox"
+          component-path: "components/terraform/s3-bucket"
+          terraform-plan-role: "arn:aws:iam::111111111111:role/acme-core-gbl-identity-gitops"
+          terraform-state-bucket: "acme-core-ue2-auto-gitops"
+          terraform-state-role: "arn:aws:iam::999999999999:role/acme-core-ue2-auto-gitops-gha"
+          terraform-state-table: "acme-core-ue2-auto-gitops"
+          aws-region: "us-east-2"
+
+```
+
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:64]:
```suggestion
```

Please note that in practice, we recommend combining this action with the [`affected-stacks`](/integrations/github-actions/affected-stacks) GitHub Action inside a matrix to plan all affected stacks in parallel.
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+* **Compatible with GitHub Cloud & Self-hosted Runners** for maximum flexibility. 
+* **Beautiful Job Summaries** don't clutter up pull requests with noisy GitHub comments
+* **100% Open Source with Permissive APACHE2 License** means you have no expensive subscriptions or long-term commitments.
+
+![Example Image](/img/github-actions/tf_plan.png)
+
+![Example Image Expanded](/img/github-actions/tf_plan_expanded.png)
+
+## Usage Example
+
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:28]: In this example, the action is triggered when certain events occur, such as a manual workflow dispatch or the opening, synchronization, or reopening of a pull request, specifically on the main branch. It specifies specific permissions related to assuming roles in AWS. Within the "plan" job, the "component" and "stack" are hardcoded. In practice, these are usually derived from another action.

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+        component: dynamodb
+        inherits:
+          - dynamodb/defaults
+      vars:
+        name: gitops-plan-storage
+        # These keys (case-sensitive) are required for the cloudposse/github-action-terraform-plan-storage action
+        hash_key: id
+        range_key: createdAt
+```
+
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:117]:
```suggestion
```

File: website/docs/integrations/github-actions/atmos-terraform-apply.md [hunk 17]
```diff
@@ -0,0 +1,64 @@
+---
+title: Atmos Terraform Apply
+sidebar_position: 50
+sidebar_label: Terraform Apply
+---
+
+The Cloud Posse GitHub Action for "Atmos Terraform Apply" runs Terraform entirely from GitHub Action workflows.
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-apply.md:7]:
```suggestion
The Cloud Posse GitHub Action for "Atmos Terraform Apply" simplifies provisioning Terraform entirely within GitHub Action workflows. It makes it very easy to understand exactly what happened directly within the GitHub UI.
```

File: website/docs/integrations/github-actions/atmos-terraform-apply.md [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+
+Given any component and stack in an Atmos supported infrastructure environment, [`github-action-atmos-terraform-apply`](https://github.com/cloudposse/github-action-atmos-terraform-apply) will retrieve an existing Terraform [planfile](https://developer.hashicorp.com/terraform/tutorials/automation/automate-terraform) from a given S3 bucket using metadata stored inside a DynamoDB table, run `atmos terraform apply` with that planfile, and format the Terraform Apply result as part of a [GitHub Workflow Job Summary](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/).
+
+This action is intended to be used together with [Atmos Terraform Plan](/integrations/github-actions/atmos-terraform-plan)
+
+## Features
+
+This GitHub Action incorporates superior GitOps support for Terraform by utilizing the capabilities of Atmos, enabling efficient management of large enterprise-scale environments.
+
+* **Implements Native GitOps** with Atmos and Terraform
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-apply.md:17]:
```suggestion
* **Implements Native GitOps** with Atmos and Terraform tightly integrated with GitHub UI
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 19]
```diff
@@ -0,0 +1,125 @@
+---
+title: Atmos Terraform Plan
+sidebar_position: 40
+sidebar_label: Terraform Plan
+---
+
+The Cloud Posse GitHub Action for "Atmos Terraform Plan" runs Terraform entirely from GitHub Action workflows.
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:7]:
```suggestion
The Cloud Posse GitHub Action for "Atmos Terraform Plan" simplifies provisioning Terraform from within GitHub using workflows. Understand precisely what to expect from running a `terraform plan` from directly within the GitHub UI for any Pull Request.
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md (hunk 15 above)
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:28]:
```suggestion
In this example, the action is triggered when certain events occur, such as a manual workflow dispatch or the opening, synchronization, or reopening of a pull request, specifically on the main branch. It specifies specific permissions related to assuming roles in AWS. Within the "plan" job, the "component" and "stack" are hardcoded. In practice, these are usually derived from another action. 
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 20]
```diff
@@ -0,0 +1,125 @@
...
+
+Given any component and stack in an Atmos supported infrastructure environment, [`github-action-atmos-terraform-plan`](https://github.com/cloudposse/github-action-atmos-terraform-plan) will run `atmos terraform plan`, generate a Terraform [planfile](https://developer.hashicorp.com/terraform/tutorials/automation/automate-terraform), store this planfile in an S3 Bucket with metadata in DynamodDB, and finally format the Terraform Plan result as part of a [GitHub Workflow Job Summary](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/).
+
+This action is intended to be used with [Atmos Terraform Apply](/integrations/github-actions/atmos-terraform-apply)
```
Comment (osterman) [inline website/docs/integrations/github-actions/atmos-terraform-plan.md:11]:
```suggestion
This action is intended to be used together with [Atmos Terraform Apply](/integrations/github-actions/atmos-terraform-apply) and [Atmos Affected Stacks](/integrations/github-actions/affected-stacks).
```

File: website/docs/integrations/github-actions/atmos-terraform-plan.md [hunk 21]
```diff
@@ -0,0 +1,125 @@
...
+  terraform:
+    # S3 Bucket for storing Terraform Plans
+    gitops/s3-bucket:
+      metadata:
+        component: s3-bucket
+        inherits:
+          - s3-bucket/defaults
+      vars:
+        name: gitops-plan-storage
+        allow_encrypted_uploads_only: false
```
[the remark this answers is not in the document]
Comment (milldr) [reply website/docs/integrations/github-actions/atmos-terraform-plan.md:100]: this planfiles are not encrypted when push/pulled with https://github.com/cloudposse/github-action-terraform-plan-storage

[the remark this answers is not in the document]
Comment (osterman) [reply website/docs/integrations/github-actions/atmos-terraform-plan.md:100]: From what I can tell, it's using SSE (Server Side Encryption)

https://github.com/cloudposse/github-action-terraform-plan-
storage/blob/68f48c4cab5d6163d634df904ef654b7e69c4e1f/src/lib/s3/s3.ts#L17-L22

[the remark this answers is not in the document]
Comment (osterman) [reply website/docs/integrations/github-actions/atmos-terraform-plan.md:100]: We will resolve this in a follow up PR.

**Review by osterman — APPROVED** [review]

**Pull request merged**